### DISCUSSION TITLE
3D polygon in oldfashioned context

### DISCUSSION THREAD

**Discussion by jbigdata-git** [openlayers/ol-cesium#1143] [Ideas]
Hi,

I use OLCS in an old-fashioned context : pure HTML and JS.

I encounter a first problem with the rendering. As you can see in screenshot 00 if I use two DIV to render, one 2D and one 3D, like the example, everything goes well.

![OLCS-00](https://github.com/openlayers/ol-cesium/assets/34024999/a83ccc74-21a2-482e-a2de-cdd168981265)

On the other hand, if I delete DIV for 2D at this point no more layers are displayed (image 01). I found a workaround by "hiding" a _DIV_ of size 1x1, I did a lot of different tests, but in the absence of the 2D _DIV_, nothing is displayed in the 3D _DIV_.

![OLCS-01](https://github.com/openlayers/ol-cesium/assets/34024999/94741c8b-7460-4123-ab7a-4eeb8f0a3a88)

Do you have any advice for me to resolve this problem properly?

Back to the main subject

I want to make 3D renderings of geojson, everything goes well with the use of the _olcs_extutded_height_ property. I would like to know if there is a way to offset the height of the base of the polygon (image 02). Some kind of _**ocls_extruded_base**_ property: The ground is at 0, the polygon starts at the _ocls_extruded_base_ height and stops at the _olcs_extruded_height_ as is currently the case.

![OLCS-02](https://github.com/openlayers/ol-cesium/assets/34024999/2f26f891-faf7-47c3-907b-5f13d6c4fdfa)

**Comment by gberaudo** [comment]
Hi @jbigdata-git,

You say you are using the oldfashioned way.
This way of doing is deprecated and will be removed in next release.
I have no information about how people use the lib. Could you please let me know:
- why you are using the oldfashioned ways?
- can you move to using the ES modules? Or if there is a blocker.

What version of ol-cesium are you using?

Regarding `ocls_extruded_base`, yes it should be possible to add this to ol-cesium.
I don't know if it is a usual use-case though. Why do you add this offset?
Also, are you doing it atop some terrain or on the ellisoid?

**Comment by jbigdata-git** [comment]
Thanks for your return.

I will try to answer your questions.

I use the _Oldfashioned_ way because it’s more efficient for me. From my experience for certain applications integrating heterogeneous components which must interact if we want to be in control of all interactions, it is uninteresting to remain low level: HTML and pure JS. My opinion is based, among other things, on the creation of this application: [AirMeteo.fr](https://AirMeteo.fr).
I therefore prefer to stay low level and above all I have to stay in an HTTP server.

I don't know if I answered your question, but if I can give an opinion it's a real shame to eliminate this simple and efficient way of using your components which corresponds to a need.

I use the _openlayers-ol-cesium-d1fb3b6.zip_ archive that I retrieved on January 6. I see there is a new version (_openlayers-ol-cesium-a64440c.zip_) and those files in OFashioned way:

- https://cdn.jsdelivr.net/npm/ol@v8.2.0/ol.css
- ./libs/openlayers-ol-cesium-d1fb3b6/olcs.css
- https://cdn.jsdelivr.net/npm/ol@v8.2.0/dist/ol.js
- ./libs/openlayers-ol-cesium-d1fb3b6/olcesium.js
- ./libs/openlayers-ol-cesium-d1fb3b6/_common.js
-  https://cdn.jsdelivr.net/npm/cesium@1.113.0/Build/Cesium/Cesium.js

About **ocls_extruded_base** or **ocls_extruded_offset**.
I'm benchmarking a full opensource version of a data analysis application created for the [French government's opendata](https://www.data.gouv.fr/fr/datasets/chiffres-departementaux-mensuels-relatifs-aux-crimes-et-delits-enregistres-par-les-services-de-police-et-de-gendarmerie-depuis-janvier-1996/#/community-reuses) platform: [4001 Explorer](https://data31tech.com/webapps/45-4K1Explorer/).

On this application you can see a use case of the desired offset, if necessary I can detail more, I have put a screenshot too. You can also access the new [Cesium version](https://data31tech.com/webapps/45-4K1Explorer/index2.html) currently in development for which I am trying to have the same level of functionality.

If I may allow myself to seek your expertise again, I am also missing the selection at the layer for which I made a post "_GeoJSON layer highlight, mousenter, mousover_”. and other points which are more detailed that I will see later, for example a setting to tilt the map at the initial display.

![4001Explorer-4](https://github.com/openlayers/ol-cesium/assets/34024999/545e216c-392f-428a-a212-bc577f9834d4)
![OLCS-4001-0](https://github.com/openlayers/ol-cesium/assets/34024999/2c149fff-de33-4644-b163-89c3658e75ef)

[JB](http://data31tech.com).
